File: RatFinder/app/src/main/java/models/Sighting.java
package models;

import java.util.HashMap;
import java.util.LinkedHashMap;

import edu.gatech.krao9.ratfinder.Rat;

public class Sighting {
    private String uniqueKey;
    private String createdDate;
    private String locationType;
    private String incidentZip;
    private String incidentAddress;
    private String city;
    private String borough;
    private String latitude;
    private String longitude;

    private LinkedHashMap<String, String> children;


    public Sighting() {
        children = new LinkedHashMap<String, String>();
    }

    public Sighting(Rat rat) {
        children = new LinkedHashMap<String, String>();
        this.setUniqueKey("" + rat.getUniqueKey());
        this.setCreatedDate(rat.getCreatedDate().toString());
        this.setLocationType(rat.getLocationType());
        this.setIncidentZip("" + rat.getIncidentZip());
        this.setIncidentAddress(rat.getIncidentAddress());
        this.setCity(rat.getCity());
        this.setBorough(rat.getBorough());
        this.setLatitude("" + rat.getLatitude());
        this.setLongitude("" + rat.getLongitude());


    }


    public Sighting(String csvLine) {
        children = new LinkedHashMap<String, String>();

        String[] parts = csvLine.split(",");
        int splitSize = parts.length;


        if (splitSize > 1) { this.setUniqueKey(parts[0]); }
        if (splitSize > 1) { this.setCreatedDate(parts[1]); }
        if (splitSize > 7) { this.setLocationType(parts[7]); }
        if (splitSize > 8) { this.setIncidentZip(parts[8]); }
        if (splitSize > 9) { this.setIncidentAddress(parts[9]); }
        if (splitSize > 16) { this.setCity(parts[16]); }
        if (splitSize > 23) { this.setBorough(parts[23]); }
        if (splitSize > 49) { this.setLatitude(parts[49]); }
        if (splitSize > 50) { this.setLongitude(parts[50]); }

    }

    public void setUniqueKey(String uniqueKey) {
        this.uniqueKey = uniqueKey;
        this.children.put("Unique Key", uniqueKey);
    }

    public void setCreatedDate(String createdDate) {
        this.createdDate = createdDate;
        //this.children.put("Created Date", createdDate);
    }

    public void setLocationType(String locationType) {
        this.locationType = locationType;
        this.children.put("Location Type", locationType);
    }

    public void setIncidentZip(String incidentZip) {
        this.incidentZip = incidentZip;
        this.children.put("Incident Zip", incidentZip);
    }


    public void setIncidentAddress(String incidentAddress) {
        this.incidentAddress = incidentAddress;
        //this.children.put("Incident Address", incidentAddress);
    }

    public void setCity(String city) {
        this.city = city;
        this.children.put("City", city);
    }

    public void setBorough(String borough) {
        this.borough = borough;
        this.children.put("Borough", borough);
    }

    public void setLatitude(String latitude) {
        this.latitude = latitude;
        this.children.put("Latitude", latitude);
    }

    public void setLongitude(String longitude) {
        this.longitude = longitude;
        this.children.put("Longitude", longitude);
    }

    //makes a copy of the hashmap so we cant alter the children member of a Sighting by getting it
    public LinkedHashMap<String, String> getChildren() {
        return new LinkedHashMap<>(this.children);
    }

    public String getUniqueKey() {
        return uniqueKey;
    }
    public String getCreatedDate() {
        return createdDate;
    }
    public String getLocationType() {
        return locationType;
    }
    public String getIncidentZip() {
        return incidentZip;
    }
    public String getIncidentAddress() {
        return incidentAddress;
    }
    public String getCity() { return city; }
    public String getBorough() {
        return borough;
    }
    public String getLatitude() {
        return latitude;
    }
    public String getLongitude() {
        return longitude;
    }

}

File: RatFinder_ios/RatFinder_ios/Models/Sighting.swift
//
//  Sighting.swift
//  RatFinder_ios
//
//  Created by William Lim on 10/15/17.
//  Copyright © 2017 Kedar Rao. All rights reserved.
//

import UIKit

class Sighting {
    var createdDate: String?
    var incidentAddress: String?
    var locationType: String?
    var incidentZip: String?
    var city: String?
    var borough: String?
    var latitude: String?
    var longitude: String?
    
    init(createdDate: String?, incidentAddress: String?)
    {
        self.createdDate = createdDate
        self.incidentAddress = incidentAddress
    }
    
    init(createdDate: String?, incidentAddress: String?, locationType: String?,
         incidentZip: String?, city: String?, borough: String?, latitude: String?,
         longitude: String?)
    {
        self.createdDate = createdDate
        self.incidentAddress = incidentAddress
        self.locationType = locationType
        self.incidentAddress = incidentAddress
        self.city = city
        self.borough = borough
        self.latitude = latitude
        self.longitude = longitude
    }

//    var children = [String: String]()
//
//    init (createdDate: String, incidentAddress: String, children: [String: String]) {
//        self.createdDate = createdDate
//        self.incidentAddress = incidentAddress
//        self.children = children
//    }
//
//    init(csvLine: String) {
//        var fields = csvLine.components(separatedBy: ",")
//        self.createdDate = fields[1]
//        self.incidentAddress = fields[9].isEmpty ? "Address not recorded" : fields[9]
//
//        self.children["uniqueKey"] = fields[0].isEmpty ? "Unqiue Key not recorded" : fields[0]
//        self.children["locationType"] = fields[7].isEmpty ? "Location Type not recorded" : fields[7]
//        self.children["incidentZip"] = fields[8].isEmpty ? "Incident Zip not recorded" : fields[8]
//        self.children["city"] = fields[16].isEmpty ? "City not recorded" : fields[16]
//        self.children["borough"] = fields[23].isEmpty ? "Borough not recorded" : fields[23]
//        self.children["latitude"] = fields[49].isEmpty ? "Latitude not recorded" : fields[49]
//        self.children["longitude"] = fields[50].isEmpty ? "Longitude not recorded" : fields[50]
//
//
//
//    }
    
}


